Documentation/design/operatorgroups.md

```diff
@@ -0,0 +1,221 @@
+
+> Note: If a CSV's spec omit's an entry of `InstallModeType`, that type is considered unsupported unless support can be inferred by an existing entry that implicitly supports it.
+
+### UnsupportedOperatorGroup
+
+If a CSV's `InstallMode`s do not support the target namespace selection of the `OperatorGroup` in its namespace, the CSV will transition to a failure state with reason `UnsupportedOperatorGroup`. CSVs in a failed state for this reason will transition to pending once either the `OperatorGroups`'s target namespace selection changes to a supported configuration, or the CSV's `InstallMode`s are modified to support the `OperatorGroup`'s target namespace selection.
+
+## Target Namespace Selection
+
+Selection of the set of namespaces by specifying a label selector with the `spec.selector` field:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-group
+  namespace: my-namespace
+spec:
+  selector:
+    cool.io/prod: "true"
+```
+
+or by explicitly naming target namespaces with the `spec.targetNamespaces` field:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-group
+  namespace: my-namespace
+  targetNamespaces:
+  - my-namespace
+  - my-other-namespace
+  - my-other-other-namespace
+```
+
+> Note: If both `spec.targetNamespaces` and `spec.selector` are defined, `spec.selector` is ignored.
+
+Additionally, a _global_ `OperatorGroup` (which selects all namespaces) is specified by omitting both `spec.selector` and `spec.targetNamespaces`:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-group
+  namespace: my-namespace
+```
+
+The resolved set of selected namespaces is surfaced in an `OperatorGroup`'s `status.namespaces` field. A global `OperatorGroup`'s `status.namespace` is of length 1 and contains the empty string, `""`, which signals a consuming operator that it should watch all namespaces.
+
+> Note: The consuming operator must know to treat `""` as an all namespace configuration.
+
+## OperatorGroup CSV Annotations
+
+Member CSVs of an `OperatorGroup` get the following annotations:
+* `olm.operatorGroup=<group-name>`
+  * Contains the name of the `OperatorGroup`
+* `olm.operatorGroupNamespace=<group-namespace>`
+  * Contains the namespace of the `OperatorGroup`
+* `olm.targetNamespaces=<target-namespaces>`
+  * Contains a comma-delimited string listing the `OperatorGroup`'s target namespace selection. This annotation is projected onto the pod template of a CSV's deployments and can be consumed by a pod instance via [The Downward API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#the-downward-api)
+
+> Note: All annotations except `olm.targetNamespaces` are included with [copied CSVs](#copied-csvs). 
```

Comment:
Could mention this is to avoid leaking namespaces.

Reply:
I point that out further into the doc, but i'll stick it in here too!